newsroom/agenda/service.py

```diff
@@ -0,0 +1,118 @@
+from datetime import datetime
+
+from newsroom.auth.utils import get_user_from_request, get_company_from_request
+from newsroom.agenda.agenda import (
+    is_events_only_access,
+    build_agenda_query,
+    nested_query,
+    planning_items_query_string,
+    aggregations,
+    remove_fields,
+    PLANNING_ITEMS_FIELDS,
+)
+from newsroom.agenda.model import FeaturedResourceModel
+from newsroom.utils import get_local_date
+from newsroom.template_filters import is_admin
+
+from superdesk.flask import abort
+from superdesk.core.resources import AsyncResourceService
+from superdesk.utc import local_to_utc
+from superdesk.utils import ListCursor
+
+
+class FeaturedService(AsyncResourceService[FeaturedResourceModel]):
+    resource_name = "agenda_featured"
+
+    async def on_create(self, docs):
+        """
+        Add UTC from/to datetimes on save.
+        Problem is 31.8. in Sydney is from 30.8. 14:00 UTC to 31.8. 13:59 UTC.
+        And because we query later using UTC, we store those UTC datetimes as
+        display_from and display_to.
+        """
+        for item in docs:
+            date = datetime.strptime(item._id, "%Y%m%d")
+            item.display_from = local_to_utc(item.tz, date.replace(hour=0, minute=0, second=0))
+            item.display_to = local_to_utc(item.tz, date.replace(hour=23, minute=59, second=59))
+        await super().on_create(docs)
+
+    async def find_one_for_date(self, for_date: datetime) -> FeaturedResourceModel | None:
+        return await self.find_one(req=None, display_from={"$lte": for_date}, display_to={"$gte": for_date})
+
+    async def get_featured_stories(self, req, lookup):
+        for_date = datetime.strptime(req.args.get("date_from"), "%d/%m/%Y %H:%M")
+        offset = int(req.args.get("timezone_offset", "0"))
+        local_date = get_local_date(
+            for_date.strftime("%Y-%m-%d"),
+            datetime.strftime(for_date, "%H:%M:%S"),
+            offset,
+        )
+        featured_doc = await self.find_one_for_date(local_date)
+        return await self.featured(req, lookup, featured_doc)
+
+    async def featured(self, req, lookup, featured):
```

Comment:
Same here, figure out what the improved function signature should be, providing only the arguments you require to get the job done.

Reply:
Actually there is an issue here, this function requires the `req` to be of Eve ParsedRequest type, and not our own Request instance. This might need a re-think (or wait until the Agenda resource is updated...)